src/dbt_mcp/discovery/client.py

```diff
@@ -258,12 +260,19 @@ def fetch_models(self, model_filter: ModelFilter | None = None) -> list[dict]:
 
         return all_edges
 
-    def fetch_model_details(self, model_name: str) -> dict:
-        variables = {
-            "environmentId": self.environment_id,
-            "modelsFilter": {"identifier": model_name},
-            "first": 1,
-        }
+    def fetch_model_details(self, model_name: str, uniqueId: str = None) -> dict:
+        if uniqueId:
+            variables = {
+                "environmentId": self.environment_id,
+                "modelsFilter": {"uniqueIds": [uniqueId]},
+                "first": 1,
+            }
+        else:
+            variables = {
+                "environmentId": self.environment_id,
+                "modelsFilter": {"identifier": model_name},
+                "first": 1,
+            }
```

Comment:
Let's be diligent with our type annotations and try to avoid too much duplication.

```suggestion
    def fetch_model_details(
        self, model_name: str, unique_id: str | None = None
    ) -> dict:
        model_filters: dict[str, list[str] | str] = (
            {"uniqueIds": [unique_id]} if unique_id else {"identifier": model_name}
        )
        variables = {
            "environmentId": self.environment_id,
            "modelsFilter": model_filters,
            "first": 1,
        }
```

Reply:
Good callout. Implementing the changes @reneruck-dbt

Reply:
Changes pushed to branch @reneruck-dbt 🫡